mutiny-core/src/node.rs

```diff
@@ -569,8 +569,65 @@ impl Node {
         amount_sat: Option<u64>,
         description: String,
         route_hints: Option<Vec<PhantomRouteHints>>,
+    ) -> Result<Invoice, MutinyError> {
+        // the amount to create for the invoice whether or not there is an lsp
+        let (amount_sat, lsp_fee_sat) = if let Some(lsp) = self.lsp_client.clone() {
+            // LSP requires an amount
+            let amount_sat = amount_sat
+                .filter(|a| a > &0)
+                .ok_or(MutinyError::BadAmountError)?;
+
+            // check the fee from the LSP
+            let lsp_fee = lsp
+                .get_lsp_fee_sat(FeeRequest {
+                    pubkey: self.pubkey.to_hex(),
+                    amount_msat: amount_sat * 1000,
+                })
+                .await?;
+            let amount_minus_fee = amount_sat
+                .checked_sub(lsp_fee)
+                .ok_or(MutinyError::BadAmountError)?;
+            (Some(amount_minus_fee), Some(lsp_fee))
+        } else {
+            (amount_sat, None)
+        };
+
+        let invoice = self
+            .create_internal_invoice(amount_sat, lsp_fee_sat, description, route_hints)
+            .await?;
+
+        if let Some(lsp) = self.lsp_client.clone() {
+            self.connect_peer(PubkeyConnectionInfo::new(&lsp.connection_string)?, None)
+                .await?;
+            let lsp_invoice_str = lsp.get_lsp_invoice(invoice.to_string()).await?;
+            let lsp_invoice = Invoice::from_str(&lsp_invoice_str)?;
+
+            let invoice_network = network_from_currency(lsp_invoice.currency());
+            if !is_valid_network(invoice_network, self.network) {
+                return Err(MutinyError::IncorrectNetwork(invoice_network));
+            }
+
+            if lsp_invoice.payment_hash() != invoice.payment_hash()
+                || lsp_invoice.recover_payee_pub_key() != lsp.pubkey
+            {
+                return Err(MutinyError::InvoiceCreationFailed);
+            }
+
+            Ok(lsp_invoice)
+        } else {
+            Ok(invoice)
+        }
+    }
+
+    async fn create_internal_invoice(
+        &self,
+        amount_sat: Option<u64>,
+        fee_amount_sat: Option<u64>,
+        description: String,
+        route_hints: Option<Vec<PhantomRouteHints>>,
     ) -> Result<Invoice, MutinyError> {
         let amount_msat = amount_sat.map(|s| s * 1_000);
+        let fee_amount_msat = fee_amount_sat.map(|s| s * 1_000);
```

Comment:
You have it dividing by 1000 when we get the fee from the lsp, just to multiply it by 1000 here, should just have `get_lsp_fee_sat` return in msats instead so we don't lose precision for no reason. Can then rename to `get_lsp_fee_msat`

Reply:
fixed.